src/Passes/README.md

```diff
@@ -0,0 +1,327 @@
+# Q# Passes for LLVM
+
+This library defines [LLVM passes](https://llvm.org/docs/Passes.html) used for analysing, optimising and transforming the IR. The Q# pass library is a dynamic library that can be compiled and ran separately from the
+rest of the project code. While it is not clear whether this possible at the moment, we hope that it will be possible to write passes that enforce the [QIR specification](https://github.com/microsoft/qsharp-language/tree/main/Specifications/QIR).
+
```

Comment:
This is a bit of a nitpick, but something I do think we need to start being careful with differentiating: these are QIR Passes for LLVM, not Q# passes. Since they process LLVM that is using QIR conventions for representing quantum runtime and intrinsic capabilities, they don't rely on anything Q# specific, and should be usable on QIR from sources other than the Q# compiler.

Reply:
Yes, agreed. Even if we have passes that rely on certain restrictions having been imposed by the source language rather than being enforced across the board by the QIR spec, we should still call them QIR passes and document the restrictions it is relying on (and at some point we will need to build validators).

Reply:
I've removed any mentioning of the Q# compiler.